### ISSUE TITLE
Everything is an object

### ISSUE THREAD

**Issue by blindman2k** [cloud-annotations/training#7]
I have finally made it to the end of the process without significant errors. I have trained it with 270 images of a single object. The react app is showing everything (the background, the chair, myself) as one of these objects. What am I doing wrong?

**Comment by bourdakos1** [comment]
hey @blindman2k, how many steps did you train for?
Check the .tmp/checkpoint directory and see what the highest checkpoint was

**Comment by blindman2k** [comment]
I am not sure what is defined as "highest".

```➜  checkpoint git:(master) ls -l
total 1070568
-rw-r--r--  1 aronsteg  staff       265 12 Feb 14:08 checkpoint
drwxr-xr-x  3 aronsteg  staff       102 12 Feb 12:38 eval_0
-rw-r--r--  1 aronsteg  staff  24690736 12 Feb 14:09 events.out.tfevents.1549934877.My-iMac.local
-rw-r--r--  1 aronsteg  staff  14573163 12 Feb 12:28 graph.pbtxt
-rw-r--r--  1 aronsteg  staff  88860808 12 Feb 13:28 model.ckpt-390.data-00000-of-00001
-rw-r--r--  1 aronsteg  staff     26410 12 Feb 13:28 model.ckpt-390.index
-rw-r--r--  1 aronsteg  staff   6796164 12 Feb 13:28 model.ckpt-390.meta
-rw-r--r--  1 aronsteg  staff  88860808 12 Feb 13:38 model.ckpt-455.data-00000-of-00001
-rw-r--r--  1 aronsteg  staff     26410 12 Feb 13:38 model.ckpt-455.index
-rw-r--r--  1 aronsteg  staff   6796164 12 Feb 13:38 model.ckpt-455.meta
-rw-r--r--  1 aronsteg  staff  88860808 12 Feb 13:48 model.ckpt-520.data-00000-of-00001
-rw-r--r--  1 aronsteg  staff     26410 12 Feb 13:48 model.ckpt-520.index
-rw-r--r--  1 aronsteg  staff   6796164 12 Feb 13:48 model.ckpt-520.meta
-rw-r--r--  1 aronsteg  staff  88860808 12 Feb 13:58 model.ckpt-585.data-00000-of-00001
-rw-r--r--  1 aronsteg  staff     26410 12 Feb 13:58 model.ckpt-585.index
-rw-r--r--  1 aronsteg  staff   6796164 12 Feb 13:58 model.ckpt-585.meta
-rw-r--r--  1 aronsteg  staff  88860808 12 Feb 14:08 model.ckpt-646.data-00000-of-00001
-rw-r--r--  1 aronsteg  staff     26410 12 Feb 14:08 model.ckpt-646.index
-rw-r--r--  1 aronsteg  staff   6796164 12 Feb 14:08 model.ckpt-646.meta
-rw-r--r--  1 aronsteg  staff  27380740  2 Feb  2018 model.ckpt.data-00000-of-00001
-rw-r--r--  1 aronsteg  staff      8937  2 Feb  2018 model.ckpt.index
-rw-r--r--  1 aronsteg  staff   3006546  2 Feb  2018 model.ckpt.meta


➜  checkpoint git:(master) cat checkpoint
model_checkpoint_path: "model.ckpt-646"
all_model_checkpoint_paths: "model.ckpt-390"
all_model_checkpoint_paths: "model.ckpt-455"
all_model_checkpoint_paths: "model.ckpt-520"
all_model_checkpoint_paths: "model.ckpt-585"
all_model_checkpoint_paths: "model.ckpt-646"

**Comment by blindman2k** [comment]
It detects the skeleton but not the Memebox. It thinks me (and just about anything else) is a Memebox. I only added the skeleton today to see if it could detect something completely different ... which it did.

![screen shot 2019-02-12 at 2 21 21 pm](https://user-images.githubusercontent.com/771549/52609572-d9e1b080-2ed1-11e9-8d03-3db12bcce925.png)

**Comment by bourdakos1** [comment]
The highest is "model.ckpt-646" which means it training for 646 steps, which is a decent amount, It's detecting the "Skeleton" well, which makes me think that the training data for "Memebox" just isn't good enough, do you have some samples of what the "Memebox" training data looks like?

**Comment by blindman2k** [comment]
Its up to iteration 6515 now :)

It's 300-odd photos of the Memebox from various angles and backgrounds. None of them include my hand holding it but none of them look much like me either.

![6beb95d3-b044-4534-8d6c-db08f40ffa4f](https://user-images.githubusercontent.com/771549/52667694-ab151a00-2f65-11e9-8c8d-ebc6e742ad39.jpg)
![6cbd526e-3d48-42ac-8e9c-0cd66dff87e7](https://user-images.githubusercontent.com/771549/52667695-abadb080-2f65-11e9-96c3-c467e3737050.jpg)
![6ce0dbc4-5189-4c43-abac-f733a1e09808](https://user-images.githubusercontent.com/771549/52667696-abadb080-2f65-11e9-83a2-6b1fb74bb51c.jpg)

**Comment by bourdakos1** [comment]
Okay, Im gonna say that is the problem, the view of the training images is very different from the conditions in the webcam test. The model learns a lot from things not in the bounding boxes, if you don't show up anywhere in the training data, then the features it picks up on when training might overlap completely with feature that your body in the webcam video also posses. Also the object is very large compared to the background in the training set, whereas in the test set it is very small, so make sure you include a variety of its size relative to background in the training set

**Comment by blindman2k** [comment]
All fair. I'll extend the working set.

**Comment by bourdakos1** [comment]
When you gave it labels of yourself did it classify you as the other objects still?

**Comment by blindman2k** [comment]
Yep. Sure did.

**Comment by bourdakos1** [comment]
That’s a bit strange

**Comment by SamiiRoebuck** [comment]
Was a solution to this found? I'm having the same problem, I've trained for 8500 steps and have over 400 images but it's still classifying everything as an object

**Comment by blindman2k** [comment]
I gave up eventually.
